### ISSUE TITLE
SfMInit_ImageListing Slow

### ISSUE THREAD

**Issue by FreakTheMighty** [openMVG/openMVG#292]
I noticed that `SfMInit_ImageListing` seems very slow. It appears that most of the the time spent inside of the [`ReadImage`](https://github.com/openMVG/openMVG/blob/develop/src/software/SfM/main_SfMInit_ImageListing.cpp#L185) call. Diving deeper, there seems to be a colorspace conversion happening. The rest of the script doesn't appear to require actual image data, only image dimensions.

It seems like if we can avoid any image manipulation we could speed this up ~one billion trillion fold.

**Comment by FreakTheMighty** [comment]
With 6 images I was seeing about a 9 minute runtime. When I exit the ReadImage call early the same images complete in 1 minute. That still seems slow to me. Why should reading the image be so expensive?

**Comment by pmoulon** [comment]
Hi @FreakTheMighty 
I know that image reading is a bit slow and it's perhaps the on the fly pixel conversion...
I have not yet performed profiling...
In fact ImageListing need to know the real image dimension.
Reading the one from EXIF data is not sufficient since sometimes user resize the image but not the EXIF w,h fields.

BTW, I know that this step is slow and that's why the next version will have a image header reader:
https://github.com/openMVG/openMVG/issues/286 in order to make this step blazing fast ;-)

BTW, it seems that 9 minute for this step for 6 images is too much.
Can you profile a bit more?

Thanks for the notification.

**Comment by FreakTheMighty** [comment]
@pmoulom I think I was mistaken. I ran the profiler on a larger set of images. Sorry for the confusion.

Still, on larger sets I think performance can be improved. I'll run the profiler again give you more exact stats.

Also, I understand that the image must be read to get  reliable dimensions, but my suggestion is that the color conversion is unnecessary for this case. I was able to get a speed up from returning from the image constructor as soon as it get width and height.
